packages/world/gas-report.json

```diff
@@ -2,98 +2,92 @@
+    "functionCall": "world.installRootModule(keysInTableModule, abi.encode(tableId))",
     "gasUsed": 1032425
   },
   {
     "source": "test/KeysInTableModule.t.sol",
     "name": "change a record on a table with keysInTableModule installed",
-    "functionCall": "world.setRecord(namespace, sourceFile, keyTuple1, abi.encodePacked(val2))",
+    "functionCall": "world.setRecord(namespace, sourceName, keyTuple1, abi.encodePacked(val2))",
     "gasUsed": 30820
   },
   {
     "source": "test/KeysInTableModule.t.sol",
     "name": "delete a record on a table with keysInTableModule installed",
-    "functionCall": "world.deleteRecord(namespace, sourceFile, keyTuple1)",
+    "functionCall": "world.deleteRecord(namespace, sourceName, keyTuple1)",
     "gasUsed": 146907
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "install keys with value module",
-    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))",
-    "gasUsed": 609250
+    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(tableId))",
+    "gasUsed": 1036002
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "Get list of keys with a given value",
-    "functionCall": "bytes32[] memory keysWithValue = getKeysWithValue(world, sourceTableId, abi.encode(value1))",
-    "gasUsed": 7658
-  },
-  {
-    "source": "test/KeysWithValueModule.t.sol",
-    "name": "compute the target table selector",
-    "functionCall": "bytes32 targetTableSelector = getTargetTableSelector(MODULE_NAMESPACE, sourceTableId)",
-    "gasUsed": 2240
+    "functionCall": "bytes32[][] memory keysWithValue = getKeysWithValue(world, tableId, abi.encode(value1))",
+    "gasUsed": 8679
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "install keys with value module",
-    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))",
-    "gasUsed": 609250
+    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(tableId))",
+    "gasUsed": 1036002
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "set a record on a table with KeysWithValueModule installed",
     "functionCall": "world.setRecord(namespace, sourceName, keyTuple1, abi.encodePacked(value))",
-    "gasUsed": 165686
+    "gasUsed": 255273
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "install keys with value module",
-    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))",
-    "gasUsed": 609250
+    "functionCall": "world.installRootModule(keysWithValueModule, abi.encode(tableId))",
+    "gasUsed": 1036002
   },
   {
     "source": "test/KeysWithValueModule.t.sol",
     "name": "change a record on a table with KeysWithValueModule installed",
     "functionCall": "world.setRecord(namespace, sourceName, keyTuple1, abi.encodePacked(value2))",
-    "gasUsed": 135337
+    "gasUsed": 340354
```

Comment:
What's the trade-off space here? This is a significant bump in gas cost, and I wonder when this ends up being cheaper than the previous implementation (unbounded is bad in theory, but when does it get bad in practice?)

Reply:
Good point. It's hard to evaluate these changes if we haven't measure the full space. 

I just opened #771 which adds gas reports for setting 10, 100, and 1000 keys. The costs for the original code are:

10 keys: **134842** gas
100 keys: **192968** gas
100 keys: **790975** gas

I merged those changes into this PR, which now has costs:

10 keys: **202529** gas
100 keys: **260655** gas
1000 keys: **858704** gas

Two observations: 
- the cost for the original code grows quite slowly.
- the cost for the new code is **NOT constant** and is always more expensive.

I think this is down to missing array methods. We cannot access index `Store` arrays, meaning we have to read the entire array unbounded from memory. We also have to maintain the array length, which uses an extra storage slot.

Reply:
If those two optimisations don't significantly amortise and reduce the constant cost, this says something about how we should program!

Reply:
Update: 

I added the new array methods, and tightly packed the storage slots in the `Index`. The new method now incurs an overhead of 41593 gas over the old method when setting 10, 100, 1000 keys.